impeller/aiks/experimental_canvas.cc

```diff
@@ -247,10 +287,61 @@ bool ExperimentalCanvas::Restore() {
     }
 
     element_entity.Render(renderer_, *render_passes_.back());
+    clip_coverage_stack_.PopSubpass();
+    transform_stack_.pop_back();
+
+    // We don't need to restore clips if a saveLayer was performed, as the clip
+    // state is per render target, and no more rendering operations will be
+    // performed as the render target workloaded is completed in the restore.
+    return true;
   }
 
+  size_t num_clips = transform_stack_.back().num_clips;
   transform_stack_.pop_back();
 
+  if (num_clips > 0) {
```

Comment:
Aren't we using the new StC depth buffer for clips? I thought it didn't need a restore...?

Reply:
We need to encode the restore so that we can correctly remove certain clips (since we see pairs of clip - restore). But the entity itself doesn't do anything.

Reply:
I'm still not clear on why all of the computations here needs to be done if this entity doesn't actually do anything and is only used to balance a restore...?

Also, what is the "certain" in "certain clips"? Are there some clips that do not use the StC model of depth buffer encoding to enforce the clips?

Reply:
oh, because we also need to update the clip scissor

Reply:
Hopefully Brandon can simplify this later...